src/java.base/share/classes/jdk/internal/jimage/ImageReader.java

```diff
@@ -468,6 +468,20 @@ Node handleModulesSubTree(String name, ImageLocation loc) {
 
         Node handleResource(String name) {
             Node n = null;
+            if (!name.startsWith("/modules/")) {
+                return null;
+            }
+            // Make sure that the thing that follows "/modules/" is a module name.
+            // This will rule out paths that start as "/modules/modules/".
```

Comment:
The code change looks fine. I guess I would probably change the comment to say that it make sure that there is something ending with "/" after "/modules/" (as the code that it comments doesn't actually change for a module name).